google-cloud-datastore/lib/google/cloud/datastore/query.rb

```diff
@@ -177,22 +206,18 @@ def kind *kinds
         #
         #   tasks = datastore.run query
         #
-        def where name, operator, value
+        def where name_or_filter, operator = nil, value = nil
           @grpc.filter ||= Google::Cloud::Datastore::V1::Filter.new(
             composite_filter: Google::Cloud::Datastore::V1::CompositeFilter.new(
               op: :AND
```

Comment:
Q: Will we always have an AND composite filter for `where` ?

Reply:
Yes, the `where` clause will always apply `AND` by default, as has been the case up to this point. The Filter class has now been introduced to allow the use of `OR`.